bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java

```diff
@@ -587,14 +587,14 @@ public Response updateFirmware(
     @Path("/{thingUID}/firmware/status")
     @ApiOperation(value = "Gets thing's firmware status.")
     @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
-            @ApiResponse(code = 404, message = "Firmware status info not found.") })
+            @ApiResponse(code = 204, message = "Firmware status info not found.") })
```

Comment:
The text very much suggests a 404 - how about something along the lines of "No firmware status info provided [by this thing]"?